mono/utils/mono-threads-state-machine.c

```diff
@@ -18,49 +18,45 @@
 static int
 get_thread_state (int thread_state)
 {
-	MonoThreadStateMachine state;
-	state.raw = thread_state;
+	const MonoThreadStateMachine state = {thread_state};
 	return state.state;
 }
 
 static int
 get_thread_suspend_count (int thread_state)
 {
-	MonoThreadStateMachine state;
-	state.raw = thread_state;
+	const MonoThreadStateMachine state = {thread_state};
 	return state.suspend_count;
 }
 
 static gboolean
 get_thread_no_safepoints (int thread_state)
 {
-	MonoThreadStateMachine state;
-	state.raw = thread_state;
+	const MonoThreadStateMachine state = {thread_state};
 	return state.no_safepoints;
 }
 
-static int
+static MonoThreadStateMachine
 build_thread_state (int thread_state, int suspend_count, gboolean no_safepoints)
 {
 	g_assert (suspend_count >= 0 && suspend_count <= THREAD_SUSPEND_COUNT_MAX);
 	g_assert (thread_state >= 0 && thread_state <= STATE_MAX);
 	no_safepoints = !!no_safepoints; // ensure it's 0 or 1
 
-	MonoThreadStateMachine state;
-	/* want a predictable value for the unused bits so that
-	 * thread_state_cas doesn't fail for spurious reasons.
+	/* need a predictable value for the unused bits so that
+	 * thread_state_cas does not fail.
 	 */
-	state.raw = 0; 
+	MonoThreadStateMachine state = { 0 };
 	state.state = thread_state;
 	state.no_safepoints = no_safepoints;
 	state.suspend_count = suspend_count;
-	return state.raw;
+	return state;
 }
 
 static int
-thread_state_cas (MonoThreadStateMachine *state, int new_raw, int old_raw)
+thread_state_cas (MonoThreadStateMachine *state, MonoThreadStateMachine new_value, int old_raw)
```

Comment:
This is a good change, because it prevents the wrong form from compiling -- typesafety.